File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/member/repository/TeacherRepository.java
package b1nd.dodam.domain.rds.member.repository;

import b1nd.dodam.domain.rds.member.entity.Member;
import b1nd.dodam.domain.rds.member.entity.Teacher;
import b1nd.dodam.domain.rds.member.exception.TeacherNotFoundException;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface TeacherRepository extends JpaRepository<Teacher, Integer> {

    void deleteByMember(Member member);

    Optional<Teacher> findByMember(Member member);

    default Teacher getByMember(Member member) {
        return findByMember(member)
                .orElseThrow(TeacherNotFoundException::new);
    }

}

File: services/service-member/src/main/kotlin/com/b1nd/dodamdodam/member/domain/teacher/repository/TeacherRepository.kt
package com.b1nd.dodamdodam.member.domain.teacher.repository

import com.b1nd.dodamdodam.member.domain.member.entity.MemberEntity
import com.b1nd.dodamdodam.member.domain.teacher.entity.TeacherEntity
import org.springframework.data.jpa.repository.JpaRepository

interface TeacherRepository : JpaRepository<TeacherEntity, Long> {
    fun findByMember(member: MemberEntity): TeacherEntity?
}
